### PR TITLE
Create API to expose container blobs.

### PR DESCRIPTION
**Pull request by newswangerd**
Adds `/api/automation-hub/_ui/v1/execution-environments/repositories/<repo>/_content/images/<ref>/` which returns detailed information about the selected image manifest specified by `ref` where ref can either be the name of an image tag, such as `latest` or the manifest digest.

This view includes the config blob for the given manifest:

```
{
    "pulp_id": "fb2536d2-e0af-426d-a8bb-a4555138234f",
    "digest": "sha256:be73fdba6d4e542e5fd10abb5aef072cf2c52547ebd3e39dd72e6ca3f98c8233",
    "schema_version": 2,
    "media_type": "application/vnd.docker.distribution.manifest.v2+json",
    "config_blob": {
        "digest": "sha256:1f0815c1cb6e74ede323d39ded15fa90182a9ad5c64a3eb405c3f4297e9390e7",
        "media_type": "application/vnd.docker.image.rootfs.diff.tar.gzip",
        "data": {
            "architecture": "amd64",
            "config": {
                "Hostname": "",
                "Domainname": "",
                "User": "",
                "AttachStdin": false,
                "AttachStdout": false,
                "AttachStderr": false,
                "ExposedPorts": {
                    "5432/tcp": {}
                },
                "Tty": false,
                "OpenStdin": false,
                "StdinOnce": false,
                "Env": [
                    "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/postgresql/13/bin",
                    "GOSU_VERSION=1.12",
                    "LANG=en_US.utf8",
                    "PG_MAJOR=13",
                    "PG_VERSION=13.2-1.pgdg100+1",
                    "PGDATA=/var/lib/postgresql/data"
                ],
                "Cmd": [
                    "postgres"
                ],
                "Image": "sha256:f353e59a49dcb986888f19e8c8f5f84ce4fdcb40c20a7a99634ac1d67a3fc621",
                "Volumes": {
                    "/var/lib/postgresql/data": {}
                },
                "WorkingDir": "",
                "Entrypoint": [
                    "docker-entrypoint.sh"
                ],
                "OnBuild": null,
                "Labels": null,
                "StopSignal": "SIGINT"
            },
            "container": "9e613460869a5ea30819cc2a35969a325770644cd2ee1bf159a22dcdd45f3527",
            "container_config": {
                "Hostname": "9e613460869a",
                "Domainname": "",
                "User": "",
                "AttachStdin": false,
                "AttachStdout": false,
                "AttachStderr": false,
                "ExposedPorts": {
                    "5432/tcp": {}
                },
                "Tty": false,
                "OpenStdin": false,
                "StdinOnce": false,
                "Env": [
                    "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/postgresql/13/bin",
                    "GOSU_VERSION=1.12",
                    "LANG=en_US.utf8",
                    "PG_MAJOR=13",
                    "PG_VERSION=13.2-1.pgdg100+1",
                    "PGDATA=/var/lib/postgresql/data"
                ],
                "Cmd": [
                    "/bin/sh",
                    "-c",
                    "#(nop) ",
                    "CMD [\"postgres\"]"
                ],
                "Image": "sha256:f353e59a49dcb986888f19e8c8f5f84ce4fdcb40c20a7a99634ac1d67a3fc621",
                "Volumes": {
                    "/var/lib/postgresql/data": {}
                },
                "WorkingDir": "",
                "Entrypoint": [
                    "docker-entrypoint.sh"
                ],
                "OnBuild": null,
                "Labels": {},
                "StopSignal": "SIGINT"
            },
            "created": "2021-02-12T19:21:50.63265542Z",
            "docker_version": "19.03.12",
            "history": [
                {
                    "created": "2021-02-09T02:20:55.909736239Z",
                    "created_by": "/bin/sh -c #(nop) ADD file:d5c41bfaf15180481d8606f50799297e3f49b8a258c7c2cd988ab2bf0013272d in / "
                },
                {
                    "created": "2021-02-09T02:20:56.183301319Z",
                    "created_by": "/bin/sh -c #(nop)  CMD [\"bash\"]",
                    "empty_layer": true
                },
                {
                    "created": "2021-02-09T18:29:55.4486208Z",
                    "created_by": "/bin/sh -c set -ex; \tif ! command -v gpg > /dev/null; then \t\tapt-get update; \t\tapt-get install -y --no-install-recommends \t\t\tgnupg \t\t\tdirmngr \t\t; \t\trm -rf /var/lib/apt/lists/*; \tfi"
                },
                {
                    "created": "2021-02-09T18:29:56.489676263Z",
                    "created_by": "/bin/sh -c set -eux; \tgroupadd -r postgres --gid=999; \tuseradd -r -g postgres --uid=999 --home-dir=/var/lib/postgresql --shell=/bin/bash postgres; \tmkdir -p /var/lib/postgresql; \tchown -R postgres:postgres /var/lib/postgresql"
                },
                {
                    "created": "2021-02-09T18:29:56.667693594Z",
                    "created_by": "/bin/sh -c #(nop)  ENV GOSU_VERSION=1.12",
                    "empty_layer": true
                },
                {
                    "created": "2021-02-09T18:30:06.477456936Z",
                    "created_by": "/bin/sh -c set -eux; \tsavedAptMark=\"$(apt-mark showmanual)\"; \tapt-get update; \tapt-get install -y --no-install-recommends ca-certificates wget; \trm -rf /var/lib/apt/lists/*; \tdpkgArch=\"$(dpkg --print-architecture | awk -F- '{ print $NF }')\"; \twget -O /usr/local/bin/gosu \"https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch\"; \twget -O /usr/local/bin/gosu.asc \"https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch.asc\"; \texport GNUPGHOME=\"$(mktemp -d)\"; \tgpg --batch --keyserver hkps://keys.openpgp.org --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4; \tgpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu; \tgpgconf --kill all; \trm -rf \"$GNUPGHOME\" /usr/local/bin/gosu.asc; \tapt-mark auto '.*' > /dev/null; \t[ -z \"$savedAptMark\" ] || apt-mark manual $savedAptMark > /dev/null; \tapt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false; \tchmod +x /usr/local/bin/gosu; \tgosu --version; \tgosu nobody true"
                },
                {
                    "created": "2021-02-09T18:30:15.387558603Z",
                    "created_by": "/bin/sh -c set -eux; \tif [ -f /etc/dpkg/dpkg.cfg.d/docker ]; then \t\tgrep -q '/usr/share/locale' /etc/dpkg/dpkg.cfg.d/docker; \t\tsed -ri '/\\/usr\\/share\\/locale/d' /etc/dpkg/dpkg.cfg.d/docker; \t\t! grep -q '/usr/share/locale' /etc/dpkg/dpkg.cfg.d/docker; \tfi; \tapt-get update; apt-get install -y --no-install-recommends locales; rm -rf /var/lib/apt/lists/*; \tlocaledef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8"
                },
                {
                    "created": "2021-02-09T18:30:15.613372377Z",
                    "created_by": "/bin/sh -c #(nop)  ENV LANG=en_US.utf8",
                    "empty_layer": true
                },
                {
                    "created": "2021-02-09T18:30:20.17839891Z",
                    "created_by": "/bin/sh -c set -eux; \tapt-get update; \tapt-get install -y --no-install-recommends \t\tlibnss-wrapper \t\txz-utils \t; \trm -rf /var/lib/apt/lists/*"
                },
                {
                    "created": "2021-02-09T18:30:21.180991634Z",
                    "created_by": "/bin/sh -c mkdir /docker-entrypoint-initdb.d"
                },
                {
                    "created": "2021-02-09T18:30:22.436760016Z",
                    "created_by": "/bin/sh -c set -ex; \tkey='B97B0AFCAA1A47F044F244A07FCC7D46ACCC4CF8'; \texport GNUPGHOME=\"$(mktemp -d)\"; \tgpg --batch --keyserver ha.pool.sks-keyservers.net --recv-keys \"$key\"; \tgpg --batch --export \"$key\" > /etc/apt/trusted.gpg.d/postgres.gpg; \tcommand -v gpgconf > /dev/null && gpgconf --kill all; \trm -rf \"$GNUPGHOME\"; \tapt-key list"
                },
                {
                    "created": "2021-02-09T18:30:22.638182505Z",
                    "created_by": "/bin/sh -c #(nop)  ENV PG_MAJOR=13",
                    "empty_layer": true
                },
                {
                    "created": "2021-02-12T19:21:23.751138402Z",
                    "created_by": "/bin/sh -c #(nop)  ENV PG_VERSION=13.2-1.pgdg100+1",
                    "empty_layer": true
                },
                {
                    "created": "2021-02-12T19:21:44.860785545Z",
                    "created_by": "/bin/sh -c set -ex; \t\texport PYTHONDONTWRITEBYTECODE=1; \t\tdpkgArch=\"$(dpkg --print-architecture)\"; \tcase \"$dpkgArch\" in \t\tamd64 | arm64 | i386 | ppc64el) \t\t\techo \"deb http://apt.postgresql.org/pub/repos/apt/ buster-pgdg main $PG_MAJOR\" > /etc/apt/sources.list.d/pgdg.list; \t\t\tapt-get update; \t\t\t;; \t\t*) \t\t\techo \"deb-src http://apt.postgresql.org/pub/repos/apt/ buster-pgdg main $PG_MAJOR\" > /etc/apt/sources.list.d/pgdg.list; \t\t\t\t\t\tcase \"$PG_MAJOR\" in \t\t\t\t9.* | 10 ) ;; \t\t\t\t*) \t\t\t\t\techo 'deb http://deb.debian.org/debian buster-backports main' >> /etc/apt/sources.list.d/pgdg.list; \t\t\t\t\t;; \t\t\tesac; \t\t\t\t\t\ttempDir=\"$(mktemp -d)\"; \t\t\tcd \"$tempDir\"; \t\t\t\t\t\tsavedAptMark=\"$(apt-mark showmanual)\"; \t\t\t\t\t\tapt-get update; \t\t\tDEBIAN_FRONTEND=noninteractive \t\t\tapt-get build-dep -y \t\t\t\tpostgresql-common pgdg-keyring \t\t\t\t\"postgresql-$PG_MAJOR=$PG_VERSION\" \t\t\t; \t\t\tDEB_BUILD_OPTIONS=\"nocheck parallel=$(nproc)\" \t\t\t\tapt-get source --compile \t\t\t\t\tpostgresql-common pgdg-keyring \t\t\t\t\t\"postgresql-$PG_MAJOR=$PG_VERSION\" \t\t\t; \t\t\t\t\t\tapt-mark showmanual | xargs apt-mark auto > /dev/null; \t\t\tapt-mark manual $savedAptMark; \t\t\t\t\t\tls -lAFh; \t\t\tdpkg-scanpackages . > Packages; \t\t\tgrep '^Package: ' Packages; \t\t\techo \"deb [ trusted=yes ] file://$tempDir ./\" > /etc/apt/sources.list.d/temp.list; \t\t\tapt-get -o Acquire::GzipIndexes=false update; \t\t\t;; \tesac; \t\tapt-get install -y --no-install-recommends postgresql-common; \tsed -ri 's/#(create_main_cluster) .*$/\\1 = false/' /etc/postgresql-common/createcluster.conf; \tapt-get install -y --no-install-recommends \t\t\"postgresql-$PG_MAJOR=$PG_VERSION\" \t; \t\trm -rf /var/lib/apt/lists/*; \t\tif [ -n \"$tempDir\" ]; then \t\tapt-get purge -y --auto-remove; \t\trm -rf \"$tempDir\" /etc/apt/sources.list.d/temp.list; \tfi; \t\tfind /usr -name '*.pyc' -type f -exec bash -c 'for pyc; do dpkg -S \"$pyc\" &> /dev/null || rm -vf \"$pyc\"; done' -- '{}' +"
                },
                {
                    "created": "2021-02-12T19:21:46.007301729Z",
                    "created_by": "/bin/sh -c set -eux; \tdpkg-divert --add --rename --divert \"/usr/share/postgresql/postgresql.conf.sample.dpkg\" \"/usr/share/postgresql/$PG_MAJOR/postgresql.conf.sample\"; \tcp -v /usr/share/postgresql/postgresql.conf.sample.dpkg /usr/share/postgresql/postgresql.conf.sample; \tln -sv ../postgresql.conf.sample \"/usr/share/postgresql/$PG_MAJOR/\"; \tsed -ri \"s!^#?(listen_addresses)\\s*=\\s*\\S+.*!\\1 = '*'!\" /usr/share/postgresql/postgresql.conf.sample; \tgrep -F \"listen_addresses = '*'\" /usr/share/postgresql/postgresql.conf.sample"
                },
                {
                    "created": "2021-02-12T19:21:47.044518686Z",
                    "created_by": "/bin/sh -c mkdir -p /var/run/postgresql && chown -R postgres:postgres /var/run/postgresql && chmod 2777 /var/run/postgresql"
                },
                {
                    "created": "2021-02-12T19:21:47.258037173Z",
                    "created_by": "/bin/sh -c #(nop)  ENV PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/postgresql/13/bin",
                    "empty_layer": true
                },
                {
                    "created": "2021-02-12T19:21:47.443839312Z",
                    "created_by": "/bin/sh -c #(nop)  ENV PGDATA=/var/lib/postgresql/data",
                    "empty_layer": true
                },
                {
                    "created": "2021-02-12T19:21:48.468491254Z",
                    "created_by": "/bin/sh -c mkdir -p \"$PGDATA\" && chown -R postgres:postgres \"$PGDATA\" && chmod 777 \"$PGDATA\""
                },
                {
                    "created": "2021-02-12T19:21:48.649828197Z",
                    "created_by": "/bin/sh -c #(nop)  VOLUME [/var/lib/postgresql/data]",
                    "empty_layer": true
                },
                {
                    "created": "2021-02-12T19:21:48.888762386Z",
                    "created_by": "/bin/sh -c #(nop) COPY file:e9c9c5e19c7b014c81f4ef8bcc5c1f247c4d9b165d34d35e9a28ca5adb5e0ab3 in /usr/local/bin/ "
                },
                {
                    "created": "2021-02-12T19:21:49.850673306Z",
                    "created_by": "/bin/sh -c ln -s usr/local/bin/docker-entrypoint.sh / # backwards compat"
                },
                {
                    "created": "2021-02-12T19:21:50.047300395Z",
                    "created_by": "/bin/sh -c #(nop)  ENTRYPOINT [\"docker-entrypoint.sh\"]",
                    "empty_layer": true
                },
                {
                    "created": "2021-02-12T19:21:50.247887424Z",
                    "created_by": "/bin/sh -c #(nop)  STOPSIGNAL SIGINT",
                    "empty_layer": true
                },
                {
                    "created": "2021-02-12T19:21:50.44392209Z",
                    "created_by": "/bin/sh -c #(nop)  EXPOSE 5432",
                    "empty_layer": true
                },
                {
                    "created": "2021-02-12T19:21:50.63265542Z",
                    "created_by": "/bin/sh -c #(nop)  CMD [\"postgres\"]",
                    "empty_layer": true
                }
            ],
            "os": "linux",
            "rootfs": {
                "type": "layers",
                "diff_ids": [
                    "sha256:9eb82f04c782ef3f5ca25911e60d75e441ce0fe82e49f0dbf02c81a3161d1300",
                    "sha256:7fb152e6f40bf72363936c92c950050cca83e9576815a74c25950abf8c0d3e8f",
                    "sha256:a0f50f5ed0d0c23483492fb10a866e490c086e5e963a2d5b8344dfe7d48db89b",
                    "sha256:613e12881e916571ff4f791fdabcd6f04623fabbc287d1d4884855f2ab1804f9",
                    "sha256:c333ce7ee08dc5cac08231dc74887a6f76e26ed04e43035beebc36e91c6a5967",
                    "sha256:442199607fd2c587b745ca09735ba9c4bd532fe628f8d9b000b9b688d712a571",
                    "sha256:f6a3a878bcd14d3f01ff50faeec1965904c988381fab3fcfc6de7471eae3566c",
                    "sha256:ab6aaa4d00577357437c2749510de29183a77d2f01d4827554c7803a86a694e5",
                    "sha256:22d37041bad88c5e90fb1e0e51a7426145b71b018bdaa479bbb68b7e17dd8b7e",
                    "sha256:fe6175ce8b5d8a5e3f7124a0a4763ef75d74a7a04df419a294fd066cc7a2ff36",
                    "sha256:cc5590fb5d4987ff2c6303aeb00d7d41502980294d55ac1dac6ab15c08eb95e0",
                    "sha256:d2ff207294364fab84df840a7a6ec29a07fa1cfb80d3ddab622f3ee3eb1cf31b",
                    "sha256:0105aacd108f58e8dd2343c357f91f58ac8746a8dde33775a50dd0200c728e19",
                    "sha256:a82dda97ed06bc6078a79ed0a1dfbb63c012e20fb1638872f57b6e1fcc031f47"
                ]
            }
        }
    },
    "tags": [
        "latest"
    ],
    "pulp_created": "2021-03-17T19:47:24.384761Z",
    "layers": [
        {
            "digest": "sha256:f63c47038e663f46e2c1191c0bfe018c039361ee8bd7ffc3352a9692cc91e1e3",
            "size": 169
        },
        {
            "digest": "sha256:ebcdc659c5bf4b2ba68155c624968151cbfa99508dba74257051245d4e1fa5e0",
            "size": 9337
        },
        {
            "digest": "sha256:c8752d5b785c50549f9d1ae2b6a004258c6ceb925a852bd5c9f30acf09f3892e",
            "size": 121
        },
        {
            "digest": "sha256:77a0c198cde5ca80ffba1a109ebc345a72b15847a58f310929c7c28f5c6cc67e",
            "size": 4399
        },
        {
            "digest": "sha256:29be22cb3acc5ee06240b3a727fa9dbbbe3b492f8c71d09fd57e84b6eeb83c00",
            "size": 130
        },
        {
            "digest": "sha256:390c1c9a5ae4cb5301a81835bf1670a6114751ef4d07a7349c8b1413291746d7",
            "size": 3050
        },
        {
            "digest": "sha256:d736e67e6ac366e5f1209a1500726480888e4d78469d8e7dcadb5e6a47bbd178",
            "size": 115
        },
        {
            "digest": "sha256:5044045cf6f211ae92b3cf444652785a22d45d61057ac2c280d4028009dbea60",
            "size": 391060
        },
        {
            "digest": "sha256:564b77596399c4678ea0968348cf4d52e02e9669209e4de9de0eb1cf48749778",
            "size": 7964523
        },
        {
            "digest": "sha256:2696974e2815b47f62f4bb06900e7fa75b75a3c0874a9786e0dda00d6467d7c3",
            "size": 1419383
        },
        {
            "digest": "sha256:b4c431d00c785ececae7cb8dfa48d970c552f3dbf319a4bde1f12d1930cc237d",
            "size": 1770
        },
        {
            "digest": "sha256:40adec129f1a1bfc5a08b13816eef32abde7672f1bd4c56ad6741529728c8ac5",
            "size": 4178277
        },
        {
            "digest": "sha256:45b42c59be334ecda0daaa139b2f7d310e45c564c5f12263b1b8e68ec9e810ed",
            "size": 27095142
        },
        {
            "digest": "sha256:c0e62f172284d5acbd13ade8030dc1e703663d26d795ae4d19d8c66b3a5499b0",
            "size": 73713565
        }
    ]
}
```

### CODE REVIEW

File: galaxy_ng/app/api/ui/viewsets/execution_environment.py [hunk 1]
```diff
@@ -80,6 +80,7 @@ def get_queryset(self):
         base_path = self.kwargs["base_path"]
         repo = get_object_or_404(models.ContainerDistribution, base_path=base_path).repository
         repo_version = repo.latest_version()
+        # TODO: Fix this with get_content in pulpcore 3.11
```
Comment (rochacbruno) [inline galaxy_ng/app/api/ui/viewsets/execution_environment.py:83]: Resolving this TODO as now we have 3.11 released? or it will come in a different PR?

**Review by rochacbruno — APPROVED** [review]

**Comment by newswangerd** [comment]
@rochacbruno I'm requesting another review because I've changed things up a little bit. I got rid of the view and move the logic to `Retrieve` in the manifest viewset.

Now `/api/automation-hub/_ui/v1/execution-environments/repositories/<repo>/_content/images/<ref>/` returns the same serializer as the one used by `../images/` except it includes the config blob in the response.

File: galaxy_ng/app/api/ui/serializers/execution_environment.py [hunk 2]
```diff
@@ -118,6 +120,22 @@ def get_tags(self, obj):
...
+
+        return {
+            'digest': obj.config_blob.digest,
+            'media_type': obj.config_blob.media_type,
+            'data': config_json
+        }
+
+    # class Meta:
+    #     model = super().Meta.model
+    #     fields = super().Meta.fields + ('config_blob')
```
Comment (rochacbruno) [inline galaxy_ng/app/api/ui/serializers/execution_environment.py:136]: this lines aren't needed to include the config_blob field?

**Review by rochacbruno — APPROVED** [review]

File: galaxy_ng/app/api/ui/serializers/execution_environment.py (hunk 2 above)
Comment (newswangerd) [reply galaxy_ng/app/api/ui/serializers/execution_environment.py:136]: oops, I forgot to delete that. Thanks

File: galaxy_ng/app/api/ui/viewsets/execution_environment.py (hunk 1 above)
Comment (newswangerd) [reply galaxy_ng/app/api/ui/viewsets/execution_environment.py:96]: Done

**Review by rochacbruno — APPROVED** [review]

**Pull request merged**